hackett-lib/hackett/private/toplevel.rkt

```diff
@@ -82,8 +83,17 @@
       (syntax/loc this-syntax
         (begin form ... (@%top-interaction . form*)))]
      [expr
-      (and~>> (get-type #'expr)
-              apply-current-subst
-              τ->string
-              (printf ": ~a\n"))
-      #'(force expr)])])
+      (define type (and~>> (get-type #'expr)
+                   apply-current-subst
+                   τ->string))
+      #`(list (force expr) ': #,type)])])
```

Comment:
I'm not super happy with the format in which the value and the type are transferred to `print-type-and-value`. I currently output a specially-formatted list, which I detect in `print-type-and-value`. I have noticed that `print-type-and-value` is sometimes called on values which were not produced by `#%top-expression`, which is why I delegate to `print-value` if I don't recognize the format. Obviously, if one of those mysterious other sources of values were to ever send a list which happens to have the right format, we would print the wrong thing. Those values are usually `#<void>` though, so I'm not super worried about that.